### ISSUE TITLE
Order of using directives should reset at preprocessor directives

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1161]
The following block results in warnings regarding the order of preprocessor directives:

```csharp
using System;
using Microsoft.VisualStudio;
using MyList = System.Collections.Generic.List<int>;

#if FORVS10 || FORVS12 || FORVS13 || FORVS15
using Microsoft.CodeAnalysis;
#else
using Microsoft.CodeAnalysis;
#endif
```

However, each time a preprocessor line is reached, the relative order of items should be reset. In other words, two `using` directives should not be compared to each other if a preprocessor line exists between them.

## Actual Result

Warnings SA1210 and SA1209 are reported.

## Expected result

No warnings.

## Impact

This affects the usability of SA1208, SA1209, SA1210, SA1211, SA1216, and SA1217.

**Comment by Noryoko** [comment]
Grabbing this.
